### PR TITLE
Update mongodb panache docs for annotationProcessors

### LINKED ISSUE #24637: Using `ReactivePanacheMongoEntity` creates `io.quarkus.bootstrap.classloading.QuarkusClassLoader` error
### Describe the bug

Working with [mongo reactive entities with panache](https://quarkus.io/guides/mongodb-panache#reactive) I'm finding that having an entity class that extends `ReactivePanacheMongoEntity` blows up with

```
java.lang.IllegalAccessError: class org.acme.service.FruitServiceTests tried to access protected field org.acme.Fruit.name (org.acme.service.FruitServiceTests and org.acme.Fruit are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @23bd2f6e)
```

under certain conditions.

### Expected behavior

I would expect everything to just work fine.

### Actual behavior

When running `./mvnw clean verify` on the attached reproducer my tests fail with 

```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 18.559 s <<< FAILURE! - in org.acme.service.FruitServiceTests
[ERROR] org.acme.service.FruitServiceTests.getAllFruits  Time elapsed: 0.734 s  <<< ERROR!
java.lang.IllegalAccessError: class org.acme.service.FruitServiceTests tried to access protected field org.acme.Fruit.name (org.acme.service.FruitServiceTests and org.acme.Fruit are in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @23bd2f6e)
        at org.acme.service.FruitServiceTests.createFruit(FruitServiceTests.java:54)
        at org.acme.service.FruitServiceTests.getAllFruits(FruitServiceTests.java:32)

2022-03-30 08:19:44,716 INFO  [io.quarkus] (main) Quarkus stopped in 0.054s
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   FruitServiceTests.getAllFruits:32->createFruit:54 IllegalAccess class org.acme...
[INFO] 
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

### How to Reproduce?

Reproducer: 
[mongo-reactive-with-panache.zip](https://github.com/quarkusio/quarkus/files/8380445/mongo-reactive-with-panache.zip)

1. Download the attached reproducer
2. Run `./mvnw clean verify`
3. See the error
4. Go into `pom.xml` and comment out the `<annotationProcessorPaths>` section inside the `maven-compiler-plugin` configuration
5. Re-run `./mvnw clean verify`
6. Tests run to success

### Output of `uname -a` or `ver`

```
Darwin edeandre-mac 21.4.0 Darwin Kernel Version 21.4.0: Mon Feb 21 20:34:37 PST 2022; root:xnu-8020.101.4~2/RELEASE_X86_64 x86_64
```

### Output of `java -version`

```
openjdk version "11.0.14" 2022-01-18
OpenJDK Runtime Environment Temurin-11.0.14+9 (build 11.0.14+9)
OpenJDK 64-Bit Server VM Temurin-11.0.14+9 (build 11.0.14+9, mixed mode)
```

### GraalVM version (if different from Java)

```
openjdk version "11.0.14" 2022-01-18
OpenJDK Runtime Environment GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05)
OpenJDK 64-Bit Server VM GraalVM CE 22.0.0.2 (build 11.0.14+9-jvmci-22.0-b05, mixed mode, sharing)
```

### Quarkus version or git rev

`2.7.5.Final`

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.8.4-bin/52ccbt68d252mdldqsfsn03jlf/apache-maven-3.8.4
Java version: 11.0.14, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/11.0.14-tem Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "12.3", arch: "x86_64", family: "mac"
```

### Additional information

I did some troubleshooting on my own. I found that it has something to do with the `<annotationProcessorPaths>` in the `maven-compiler-plugin` configuration within `pom.xml`. 

This simple reproducer uses the Mapstruct annotation processor. This sample itself does not use Mapstruct, but I built this reproducer off of another more complicated application I have that does use Mapstruct.

The problem isn't necessarily Mapstruct though. Just having something in `<annotationProcessorPaths>` makes it blow up. You can comment out the Mapstruct annotation processor & instead un-comment the Lombok annotation processor and see the same results.

What's also interesting is if I switch and use the repository pattern rather than the active record pattern - I no longer have this error. If you see this reproducer that uses the repository pattern ([mongo-reactive-with-panache-repository.zip](https://github.com/quarkusio/quarkus/files/8380456/mongo-reactive-with-panache-repository.zip)) and run `./mvnw clean verify`, everything is fine.

### PR DESCRIPTION
**Pull request by edeandrea**
update mongodb panache docs with annotationProcessor info

Closes #24637

### CODE REVIEW

File: docs/src/main/asciidoc/mongodb-panache.adoc [hunk 1]
```diff
@@ -87,6 +87,37 @@ If you don't want to generate a new project, add the dependency in your build fi
 implementation("io.quarkus:quarkus-mongodb-panache")
 ----
 
+[NOTE]
+====
+If your project is already configured to use other annotation processors, you will need to additionally add the panache annotation processor:
```
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-panache.adoc:92]:
```suggestion
If your project is already configured to use other annotation processors, you will need to additionally add the Panache annotation processor:
```

File: docs/src/main/asciidoc/mongodb-panache.adoc [hunk 2]
```diff
@@ -87,6 +87,37 @@ If you don't want to generate a new project, add the dependency in your build fi
...
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+<plugin>
+    <artifactId>maven-compiler-plugin</artifactId>
+    <version>${compiler-plugin.version}</version>
+    <configuration>
+        <parameters>${maven.compiler.parameters}</parameters>
+            <annotationProcessorPaths>
```
Comment (gsmet) [inline docs/src/main/asciidoc/mongodb-panache.adoc:102]: Maybe add a line with `<!-- Your existing annotation processor... -->` and a blank line to make it clear it's an addition?

**Review by gsmet — COMMENTED** [review]
I added some small suggestions.

Thanks for using the tabs :).

File: docs/src/main/asciidoc/mongodb-panache.adoc (hunk 2 above)
Comment (edeandrea) [reply docs/src/main/asciidoc/mongodb-panache.adoc:102]: That's a good idea. I'll add it in once I get to my desk this morning.

**Comment by edeandrea** [comment]
@gsmet what about MongoDB with Panache on Kotlin (https://quarkus.io/guides/mongodb-panache-kotlin)? Is there anything that needs to be updated there? I don't have much familiarity with Kotlin and don't know whether or not this annotation processor thing should be applied there too.

**Comment by edeandrea** [comment]
I also updated based on your suggestions @gsmet . I rebased back to a single commit as well.

**Comment by gsmet** [comment]
> I don't have much familiarity with Kotlin and don't know whether or not this annotation processor thing should be applied there too.

Your guess is as good as mine :).

/cc @evanchooly

**Review by gsmet — APPROVED** [review]

**Comment by loicmathieu** [comment]
Thanks @edeandrea the same should be done for Hibernate ORM with Panache and Hibernate Reactive with Panache as they also need the annotation processor.

**Pull request merged**